File: python/sea/execute.py
import json

from squid_py.ocean.ocean import Ocean

from ..config import CONFIG_FILE
from ..input import INPUT

import logging
logging.getLogger().setLevel(logging.WARNING)

ocean = Ocean(CONFIG_FILE)
provider_address = ocean.config.get('keeper-contracts', 'provider.address')
provider_password = ocean.config.get('keeper-contracts', 'provider.password')
ocean.set_main_account(provider_address, provider_password)

did = INPUT['did']
sa_id = INPUT['serviceAgreementId']
sa_hash = INPUT['serviceAgreementHash']
sa_signature = INPUT['serviceAgreementSignature']
service_id = INPUT['serviceDefinitionId']
consumer_address = INPUT['consumerAddress']

receipt = ocean.execute_service_agreement(
    service_agreement_id=sa_id,
    service_index=service_id,
    service_agreement_signature=sa_signature,
    did=did,
    consumer_address=consumer_address,
    publisher_address=ocean.main_account.address)

print(receipt)
print("__result__{}".format(json.dumps(
    {
        'transactionHash': receipt['transactionHash'].hex(),
        'blockHash': receipt['blockHash'].hex(),
        'receiptStatus': receipt['status']
    }, indent=2, sort_keys=True)))

File: node/sea/execute.js
const { Ocean, Logger } = require('@oceanprotocol/squid')
const Web3 = require('web3')
const config = require('../config')
const input = require('../input');

(async () => {
    const ocean = await Ocean.getInstance(config)
    const web3Provider = config.web3Provider || new Web3.providers.HttpProvider(config.nodeUri)
    const web3 = new Web3(Web3.givenProvider || web3Provider)
    await web3.eth.personal.unlockAccount(config.address, config.password)

    const accounts = await ocean.getAccounts()
    const publisherAccount = accounts.filter(account => account.id === config.address)[0]

    const {
        did,
        serviceAgreementId,
        serviceDefinitionId,
        serviceAgreementSignature,
        consumerAddress
    } = input

    const serviceAgreement = await ocean.executeServiceAgreement(
        did,
        serviceDefinitionId,
        serviceAgreementId.replace('0x', ''),
        serviceAgreementSignature,
        accounts.filter(account => account.id === consumerAddress)[0],
        publisherAccount)
    Logger.log('ServiceAgreementId', serviceAgreement.getId())
})()
